CorsixTH/Lua/rooms/ward.lua

```diff
@@ -166,12 +165,14 @@ function WardRoom:commandEnteringPatient(patient)
     length = length - 1
   end
   -- New callback
-  patient.healing_required = math.random(150, 600)
+  local required_work = math.random(150, 600)
   local --[[persistable:ward_new_loop_callback]] function new_loop_callback(action)
-    patient.healing_required = patient.healing_required - self.healing_amount
-    if patient.healing_required <= 0 then -- The patient is diagnosed or healed
+    if not action.remaining_work and required_work then
+      action.remaining_work = required_work
+    end
+    action.remaining_work = action.remaining_work - self.healing_amount
```

Comment:
When you write an `if`, please also carefully consider _all_ possible cases.

1. `not action.remaining_work` holds and `required_work` holds, 
2. `not action.remaining_work` fails and `required_work` holds,
3. `not action.remaining_work` fails and `required_work` fails, and
4. `not action.remaining_work` holds and `required_work` fails.

In case 1, the `then` code is performed, giving `action.remaining_work` a value, and the code works.
In case 2, `action.remaining_work` already has a value, and the code works.
In case 3, `action.remaining_work` already has a value, and the code works.
In case 4, `action.remaining_work` will not get a value, and the **program crashes** at this line.

Reply:
This is incidentally not the way I would recommend setting the value; you're checking something every loop that only needs to be initialized once. What I would suggest instead is creating the action, setting your property on the action, and then queuing it on the patient (line 180)

The same action there is passed into your callback.

Reply:
Done, if I understood.

Reply:
Yeah, that's what I meant. Only nit is that I would fully instantiate the action outside of the queueAction method now that you're starting to do it above.

```lua
local bed_action = UseObjectAction(bed)
  :setProlongedUsage(true)
  :setLoopCallback(new_loop_callback)
  :setAfterUse(after_use)
bed_action.remaining_work = math.random(150, 600)
patient.queueAction(bed_action)
```

Reply:
One other nit is the word new. If we ever replace it what does the replacement get called, new new? for the persist string I suggest the mundane ward_loop_callback_2, and for the actual function, loop_callback is fine, the old one can just be given the anonymous name `_`